solutions: reference designs: Added CN0584 page and sub-pages

jvelasc2 opened the pull request:
## Type
- [x] Documentation
- [ ] Bug fix
- [ ] New feature
- [ ] Continuous integration

## Checklist
- [x] I have performed a self-review of changes
- [x] I have followed the guidelines:
  * https://analogdevicesinc.github.io/documentation/contributing/docs_guidelines.html
  * https://analogdevicesinc.github.io/doctools/docs_guidelines/index.html
  * https://analogdevicesinc.github.io/documentation/contributing/creating_new_pages.html#documentation-structure
- [x] I have used the appropriate roles and directives:
  * https://analogdevicesinc.github.io/doctools/docs_guidelines/roles.html
  * https://analogdevicesinc.github.io/doctools/docs_guidelines/directives.html
- [x] I ensured that binary data was committed to git lfs:
  * https://analogdevicesinc.github.io/documentation/contributing/creating_new_pages.html#adding-images-and-other-binary-files
- [x] I have built the pages I edited or created, using either `make html` or `adoc serve`:
  * https://analogdevicesinc.github.io/doctools/cli.html#serve
- [x] I have signed-off my commits and believe my contribution improves the repository.

thorenscientific commented:
@jvelasc2  I'm traveling this week, but will review as soon as I can. Thanks for putting this together!

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/HDL-project/index.rst:
```diff
@@ -0,0 +1,100 @@
+Introduction
+------------
+
+This page describes the HDL project for 
+:dokuwiki:`CN0584 Precision Low Latency Development Kit </resources/eval/user-guides/circuits-from-the-lab/cn0584>`
+and :dokuwiki:`CN0585 Quad Channel, Low Latency, Data Acquisition and Signal Generation Module </resources/eval/user-guides/circuits-from-the-lab/cn0585>`.
```
@jvelasc2  these are referring to the wiki pages, these should point to the soon to be in existence GitHub docs pages.

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/HDL-project/index.rst:
```diff
@@ -0,0 +1,100 @@
+
+- :adi:`CN0584 Precision Low Latency Development Kit <cn0584>`
+- :adi:`CN0585 Quad Channel, Low Latency, Data Acquisition and Signal Generation Module <cn0585>`
+- :dokuwiki:`AXI_AD3552R IP core </resources/fpga/docs/axi_ad3552r>`
+- :dokuwiki:`AXI_LTC2387 IP core </resources/fpga/docs/axi_ltc2387>`
+- :dokuwiki:`AD3552R Dual Channel, 16-Bit, 33 MUPS, Multispan, Multi-IO SPI DAC Linux device driver <resources/tools-software/linux-drivers/iio-dac/axi-ad3552r>`
+- :dokuwiki:`LTC2387 SAR ADC IIO Linux Driver <resources/tools-software/linux-drivers/iio-adc/ltc2387>`
```
@jvelasc2  these should point to docs in https://analogdevicesinc.github.io/hdl/library/index.html
@gastmaier should we point directly to the URL for the core documentation, or is there a role? (Just thinking ahead if stuff moves around)

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+       (Note: Figure 4 features EVAL-CN0585-FMCZ RevA board. Current LLDK system has RevB board shown in
+       Figure 5, same connectors and functionalities, only different in USB-C power supply.)
+  - :adi:`EVAL-CN0584-EBZ <CN0584>`
+  - `ZedBoard <https://www.avnet.com/wps/portal/us/products/avnet-boards/avnet-board-families/zedboard/>`__
```
@jvelasc2  can you change the URL to https://digilent.com/reference/programmable-logic/zedboard/start

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+Setting up the Hardware
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Prepare the `ZedBoard <https://www.avnet.com/wps/portal/us/products/avnet-boards/avnet-board-families/zedboard/>`__.
```
Change to https://digilent.com/reference/programmable-logic/zedboard/start

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/HDL-project/index.rst:
```diff
@@ -0,0 +1,100 @@
+
+The DAC data may be sourced from an internal data generator (DDS or pattern),
+from the external DDR via DMA or from the ADC. The source control selection can
+be made by updating the :dokuwiki:`REG_CHAN_CNTRL_7 (DAC Channel) </resources/fpga/docs/hdl/regmap#:~:text=DAC%20Channel%20(axi_ad*)>`
```
@jvelasc2 I THINK this would be the new URL: https://analogdevicesinc.github.io/hdl/user_guide/ip_cores/axi_dac/index.html#dac-channel
@gastmaier does this make sense?

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/HDL-project/index.rst:
```diff
@@ -0,0 +1,100 @@
+The :dokuwiki:`carriers (abbreviations can be found here) </resources/fpga/docs/hdl#list_of_carriers_supported>` 
+are commonly available FPGA evaluation boards.
+
+The :dokuwiki:`HDL user guide </resources/fpga/docs/hdl>` contains all the documentation, build instructions and register map tables.
```
https://analogdevicesinc.github.io/hdl/user_guide/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/HDL-project/index.rst:
```diff
@@ -0,0 +1,100 @@
+Help & Support
+----------------
+
+The :dokuwiki:`carriers (abbreviations can be found here) </resources/fpga/docs/hdl#list_of_carriers_supported>` 
```
Looks like this page:
https://wiki.analog.com/resources/fpga/docs/hdl#list_of_carriers_supported
doesn't have an equivalent yet, leave as-is

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+  - Host PC (Windows or Linux)
+  - A UART terminal if need to access Linux system on the ZedBoard 
+    (Putty/TeraTerm/Minicom, etc.), Baud rate 115200 (8N1)
+  - :dokuwiki:`IIO Oscilloscope </resources/tools-software/linux-software/iio_oscilloscope>`
```
https://analogdevicesinc.github.io/documentation/software/iio-oscilloscope/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+  - A UART terminal if need to access Linux system on the ZedBoard 
+    (Putty/TeraTerm/Minicom, etc.), Baud rate 115200 (8N1)
+  - :dokuwiki:`IIO Oscilloscope </resources/tools-software/linux-software/iio_oscilloscope>`
+  - :dokuwiki:`Kuiper Linux Image </resources/tools-software/linux-software/kuiper-linux>`
```
https://analogdevicesinc.github.io/documentation/linux/kuiper/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+To boot the ZedBoard and control the :adi:`EVAL-CN0585-FMCZ <CN0585>`, you will
+need to install ADI Kuiper Linux on an SD card. Complete instructions, including
+where to download the SD card image, how to write it to the SD card, and how to
+configure the system are provided on the :dokuwiki:`Kuiper Linux page </resources/tools-software/linux-software/kuiper-linux>`.
```
https://analogdevicesinc.github.io/documentation/linux/kuiper/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+
+The user needs to supply a **Uniform Resource Identifier (URI)** which will be
+used in the context creation. To get the URI, use the command iio_info in the
+terminal. The :dokuwiki:`iio_info </resources/tools-software/linux-software/libiio/iio_info>`
```
https://analogdevicesinc.github.io/documentation/software/libiio/cli.html#iio-info

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+IIO Oscilloscope
+~~~~~~~~~~~~~~~~
+
+:dokuwiki:`IIO Oscilloscope </resources/tools-software/linux-software/iio_oscilloscope>`
```
https://analogdevicesinc.github.io/documentation/software/iio-oscilloscope/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+~~~~~~~~~
+
+The CN0584 can be interfaced to Python using the
+:dokuwiki:`PyADI-IIO </resources/tools-software/linux-software/pyadi-iio>`. 
```
https://analogdevicesinc.github.io/documentation/software/pyadi-iio/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+
+Follow the step-by-step procedure on how to install, configure, and set up
+PyADI-IIO and install the necessary packages/modules needed by referring to
+:dokuwiki:`PyADI-IIO </resources/tools-software/linux-software/pyadi-iio>`.
```
https://analogdevicesinc.github.io/documentation/software/pyadi-iio/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+   
+   Github link for the Matlab sample script: :git-repo:`CN0585StreamingTest.m </HighSpeedConverterToolbox/blob/cn0585_v1/test/CN0585StreamingTest.m>`
+
+The steps described in the :dokuwiki:`Analog Devices Transceiver Toolbox For MATLAB and Simulink </resources/tools-software/transceiver-toolbox>` 
```
https://analogdevicesinc.github.io/documentation/software/matlab/transceiver-toolbox/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
```diff
@@ -0,0 +1,852 @@
+The device digital interface is handled by specific device cores axi_ad35552r
+for the DAC path and axi_ltc2387 for the ADC path. The cores are programmable
+through an AXI-lite interface. Details of reference design can be found in
+:dokuwiki:`Reference Design </resources/eval/user-guides/circuits-from-the-lab/cn0585/reference_hdl>`.
```
Replace with new URL (from CN0585 PR)

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
Replace with URL to MATLAB page in this PR.

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
Replace with URL to digital template in this PR

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
https://analogdevicesinc.github.io/documentation/software/matlab/hsx-toolbox/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
https://analogdevicesinc.github.io/documentation/software/iio-oscilloscope/index.html

thorenscientific commented on docs/solutions/reference-designs/eval-cn0584-ebz/index.rst:
https://analogdevicesinc.github.io/documentation/linux/kuiper/index.html

thorenscientific commented:
@jvelasc2 , @gastmaier , a couple of other observations and questions:
Most of the wiki links have updated github docs equivalents. Is there a role for referring to docs hosted in other repos, like HDL in this case? I suppose it's more likely that stuff in the documentation repo itself will move around (reference designs, eval board docs, etc.) so absolute URLs to HDL documentation wouldn't be a problem, but what about the case where HDL docs move? (might not be a problem here, but thinking of future cases.)

There are a few zip files that may want to move somewhere else - python files that may be more appropriate in pyadi-iio, boot files that might be incorporated into Kuiper Linux or some other CI build process.

Most of the images look okay, but are probably not up to standard. Can we leave as-is for now, and work with the apps engineers to re-shoot as svg (or whatever makes the most sense)?

Lastly, the HDL project page is pretty redundant to https://analogdevicesinc.github.io/hdl/projects/cn0585/index.html. I'd say let's leave as-is for the moment, and work with the apps engineers on updating.

jvelasc2 commented:
Hi Mark @thorenscientific ,
Thank you very much for reviewing these documents. I will work on the URL cleanup and update this PR the soonest!

-- Joyce

jvelasc2 commented:
> Hi @thorenscientific
> 
> > references, absolute links
> 
> References to other repos must use [external references](https://analogdevicesinc.github.io/doctools/docs_guidelines/index.html#external-references) e.g.
> 
> ```
> :external+hdl:ref:`cn0585`.
> ```
> 
> (the title is obtained from the ref) No absolute links, always refs or external refs. It is also imporant to remember that for each repo the refs are global, so this is ok
> 
> ```
> .. _eval-cn0584-ebz digital-template:
> ```
> 
> this is not
> 
> ```
> .. _digital-template:
> ```
> 
> Because a similar page under `eval-cn0584-new` will conflict with the already taken label.
> 
> > There are a few zip files that may want to move somewhere else - python files that may be more appropriate in pyadi-iio, boot files that might be incorporated into Kuiper Linux or some other CI build process.
> 
> I agree, but for public they need to be under analog's sw-download. Since this repo uses git lfs, I am ok with them here, but I do wish we organized the blobs better, the naming is all over the place.
> 
> > Most of the images look okay, but are probably not up to standard. Can we leave as-is for now, and work with the apps engineers to re-shoot as svg (or whatever makes the most sense)?
> 
> Yes... totally agree... i hate ugly screenshots.
> 
> > Lastly, the HDL project page is pretty redundant to https://analogdevicesinc.github.io/hdl/projects/cn0585/index.html. I'd say let's leave as-is for the moment, and work with the apps engineers on updating.
> 
> Yup, removed. If I deleted content not over [cn05867](https://analogdevicesinc.github.io/hdl/projects/cn0585/index.html), please, re-add, but under the hdl repo...
